src/engine/platform/scc.cpp

```diff
@@ -161,9 +161,14 @@ void DivPlatformSCC::tick(bool sysTick) {
       chan[i].freq=parent->calcFreq(chan[i].baseFreq,chan[i].pitch,true,0,chan[i].pitch2,chipClock,CHIP_DIVIDER)-1;
       if (chan[i].freq<0) chan[i].freq=0;
       if (chan[i].freq>4095) chan[i].freq=4095;
-      rWrite(regBase+0+i*2,chan[i].freq&0xff);
-      rWrite(regBase+1+i*2,chan[i].freq>>8);
+      if (!chan[i].freqInit || regPool[regBase+0+i*2]!=(chan[i].freq&0xff)) {
```

Comment:
Hmmm... did you test for forceIns behavior? (maybe set freqInit to false there)

Reply:
Ah good point. I’ve added a `skipRegisterWrites` check when setting `freqInit`.

Reply:
I could also clear the flag in `forceIns` (it would do no harm), but I think this is enough.

Reply:
Ah I’ll just do it too. Just for good measure.